src/main/java/network/brightspots/rcv/CandidateRankingsList.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * RCTab
+ * Copyright (c) 2017-2022 Bright Spots Developers.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Purpose: Low-Memory container for a single cast vote record
+ * Design: Assumes valid ballots but supports overvotes and skipped rankings.
+ * Conditions: Always.
+ * Version history: see https://github.com/BrightSpots/rcv.
+ */
+
+
+package network.brightspots.rcv;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import javafx.util.Pair;
+
+class CandidateRankingsList implements Iterable<Pair<Integer, CandidatesAtRanking>> {
+  class CandidateRankingsListIterator implements Iterator<Pair<Integer, CandidatesAtRanking>> {
+    private int iteratorIndex = 0;
+
+    private CandidateRankingsList list;
+
+    CandidateRankingsListIterator(CandidateRankingsList list) {
+      this.list = list;
+    }
+
+    public boolean hasNext() {
+      if (iteratorIndex < list.rankings.length) {
+        return true;
+      }
+      return false;
+    }
+
+    public Pair<Integer, CandidatesAtRanking> next() {
+      if (iteratorIndex == list.rankings.length) {
+        throw new NoSuchElementException();
+      }
+
+      iteratorIndex++;
+      while (!hasRankingAt(iteratorIndex)) {
+        iteratorIndex++;
+        assert iteratorIndex <= list.rankings.length;
+      }
+
+      // Note: round numbers are 1-indexed externally, 0-indexed internally,
+      // thus why we return a different value than what we index into here
+      return new Pair<>(iteratorIndex, list.rankings[iteratorIndex - 1]);
+    }
+
+    public void remove() {
+      throw new UnsupportedOperationException();
+    }
+  }
+
+  private CandidatesAtRanking[] rankings;
+  private int numRankings;
+  private int iteratorIndex = 0;
+
+  CandidateRankingsList(List<Pair<Integer, String>> rawRankings) {
+    if (rawRankings.isEmpty()) {
+      this.rankings = new CandidatesAtRanking[0];
+      numRankings = 0;
+      return;
+    }
+
+    rawRankings.sort(Comparator.comparingInt(Pair::getKey));
+
+    // Initialize up to maxRankings, leaving empty arrays for any undervotes
+    int minRanking = rawRankings.get(0).getKey();
+    if (minRanking <= 0) {
+      Logger.severe("Rankings must start at 1, but you have a ranking at %d", minRanking);
+      // TODO should I throw from here or not?
```

Comment:
Since this should never happen, I don't want to litter the code with `throws` and `try/catch`. Is it okay to throw an unchecked exception here (`RuntimeException`?) Not sure what the best practices are in this regard.